### PR TITLE
Annotations in extras

### PR DESCRIPTION
**Pull request by sswistun-vl**
Added extracting annotations and storing it in extras.

### CODE REVIEW

File: plugins/base/src/main/kotlin/translators/descriptors/DefaultDescriptorToDocumentableTranslator.kt [hunk 1]
```diff
@@ -457,46 +477,53 @@ private class DokkaDescriptorVisitor( // TODO: close this class and make it priv
...
         ExtraModifiers.DYNAMIC.takeIf { isDynamic() },
         ExtraModifiers.CONST.takeIf { isConst },
         ExtraModifiers.LATEINIT.takeIf { isLateInit },
         ExtraModifiers.STATIC.takeIf { isJvmStaticInObjectOrClassOrInterface() },
         ExtraModifiers.EXTERNAL.takeIf { isExternal },
         ExtraModifiers.OVERRIDE.takeIf { DescriptorUtils.isOverride(this) }
-    ).toContainer()
+    ).toProperty()
+
+    fun List<ExtraModifiers>.toProperty() =
```
Comment (kamilok1965) [inline plugins/base/src/main/kotlin/translators/descriptors/DefaultDescriptorToDocumentableTranslator.kt]: Should be at least private or inlined as `.let(::AdditionalModifiers)`

File: plugins/base/src/main/kotlin/translators/descriptors/DefaultDescriptorToDocumentableTranslator.kt [hunk 2]
```diff
@@ -457,46 +477,53 @@ private class DokkaDescriptorVisitor( // TODO: close this class and make it priv
...
+            annotation.allValueArguments.map { (k, v) -> k.asString() to v.value.toString() }.toMap()
+        )
+    }.let { Annotations(it) }
 
-    inline fun <reified D : Documentable> List<ExtraModifiers>.toContainer(
-        container: PropertyContainer<D> = PropertyContainer.empty()
-    ): PropertyContainer<D> =
-        container + AdditionalModifiers(this)
+    inline fun <reified D : Documentable> toContainer(vararg extras: ExtraProperty<D>): PropertyContainer<D> =
+        extras.fold(PropertyContainer.empty(), { pc, p -> pc + p })
```
Comment (kamilok1965) [inline plugins/base/src/main/kotlin/translators/descriptors/DefaultDescriptorToDocumentableTranslator.kt]: Please create a method in the `PropertyContainer` like `addAll` or something similar

File: plugins/base/src/test/kotlin/model/ClassesTest.kt [hunk 3]
```diff
@@ -401,9 +401,25 @@ class ClassesTest : AbstractModelTest("/src/main/kotlin/classes/Test.kt", "class
         }
     }
 
+    @Test fun annotatedClass() {
+        inlineModelTest(
+            """
+                |@Suppress("abc") class Foo() {}
+                """
+        ) {
```
Comment (kamilok1965) [inline plugins/base/src/test/kotlin/model/ClassesTest.kt]: Formatting here looks bad

File: plugins/base/src/main/kotlin/translators/descriptors/DefaultDescriptorToDocumentableTranslator.kt [hunk 4]
```diff
@@ -457,46 +477,53 @@ private class DokkaDescriptorVisitor( // TODO: close this class and make it priv
...
+    ).toProperty()
+
+    fun List<ExtraModifiers>.toProperty() =
+        AdditionalModifiers(this)
+
+    fun DeclarationDescriptor.getAnnotations() = annotations.map { annotation ->
+        Annotations.Annotation(annotation?.let { it.annotationClass as DeclarationDescriptor }?.let { DRI.from(it) },
+            annotation.allValueArguments.map { (k, v) -> k.asString() to v.value.toString() }.toMap()
+        )
+    }.let { Annotations(it) }
```
Comment (kamilok1965) [inline plugins/base/src/main/kotlin/translators/descriptors/DefaultDescriptorToDocumentableTranslator.kt]: `.let(::Annotations)` should also work I think and looks better

File: plugins/base/src/main/kotlin/translators/descriptors/DefaultDescriptorToDocumentableTranslator.kt [hunk 5]
```diff
@@ -200,7 +220,7 @@ private class DokkaDescriptorVisitor( // TODO: close this class and make it priv
             companion = descriptor.companion(driWithPlatform),
             supertypes = PlatformDependent.from(platformData, info.supertypes),
             platformData = listOf(platformData),
-            extra = descriptor.additionalExtras()
+            extra = toContainer(descriptor.additionalExtras(), descriptor.getAnnotations())
```
Comment (kamilok1965) [inline plugins/base/src/main/kotlin/translators/descriptors/DefaultDescriptorToDocumentableTranslator.kt]: You can write it as `PropertyContainer.empty() + descriptor.additionalExtras() + descriptor.getAnnotations()` or create a `addAll()` method

**Comment by kamilok1965** [comment]
Also commit names do not conform to commit naming guidelines

File: plugins/base/src/main/kotlin/translators/descriptors/DefaultDescriptorToDocumentableTranslator.kt (hunk 2 above)
Comment (sswistun-vl) [reply plugins/base/src/main/kotlin/translators/descriptors/DefaultDescriptorToDocumentableTranslator.kt]: Created.

File: plugins/base/src/test/kotlin/model/ClassesTest.kt (hunk 3 above)
Comment (sswistun-vl) [reply plugins/base/src/test/kotlin/model/ClassesTest.kt]: Changed.

File: plugins/base/src/main/kotlin/translators/descriptors/DefaultDescriptorToDocumentableTranslator.kt (hunk 4 above)
Comment (sswistun-vl) [reply plugins/base/src/main/kotlin/translators/descriptors/DefaultDescriptorToDocumentableTranslator.kt]: Changed.

File: plugins/base/src/main/kotlin/translators/descriptors/DefaultDescriptorToDocumentableTranslator.kt (hunk 5 above)
Comment (sswistun-vl) [reply plugins/base/src/main/kotlin/translators/descriptors/DefaultDescriptorToDocumentableTranslator.kt]: Added.

**Pull request closed without merging**